src/lib.rs

```diff
@@ -30,11 +30,29 @@ pub fn change_dir() {
 pub fn set_commands(debug: bool, builder: &mut ArgsBuilder, matches: &ArgMatches) {
     let mut commands: Vec<String> = Vec::new();
 
+    // --features are injected at start of every cargo invocation
```

Comment:
And this isn't either, given we now push at the _end_ of the commands

Reply:
...which... we can't do, as per https://github.com/passcod/cargo-watch/issues/151#issuecomment-644426718

Consider `cargo watch --features foo -x 'run -- --options=for --binary'`: the code as it is now would run `cargo run --options=for --binary --features foo` which would not have the desired effect.

Reply:
This would require us to split the first word off the subcommand (if any) and inject `--features foo`. Will return to the drawing board!